### DISCUSSION TITLE
HA reset the names and icons

### DISCUSSION THREAD

**Discussion by roland1970** [jpcornil-git/HA-mcp23017#23] [General]
Since the last 2 updates from HA , all the names and icons reset to standard icon en the names are button_1 ....button_2.

**Comment by jpcornil-git** [comment]
These names are not related to this integration (only default names are for entities created using a config flow and use "pin 0x<i2c_address>:<pin number>" template rather than "button_<pin number>"

Are you creating your mcp23017 component from configuration.yaml (where you may use button_xx) or from config flow ?

How did you customized names and icons ?

**Comment by roland1970** [comment]
I create the mcp from configuration.yaml.
The icons and names did I customized with the settings from the entity.

**Comment by jpcornil-git** [comment]
Entities imported from configuration.yaml ("legacy" flow) are created, i.e. reset, after every HA restart while these created from config flow are just reloaded from the database (incl. customisations).

=> If you use configuration.yaml, you should probably use a customize section in it as well to customize icons and names rather than UI settings, see https://www.home-assistant.io/integrations/homeassistant/#editing-entity-settings-in-yaml

**Comment by roland1970** [comment]
Thanks , I'm going to test it

**Comment by roland1970** [comment]
I test it with the UI but after an reboot it reset all names and icons .
And in configuration.yaml if I change the name HA accept it if I save but
after the reboot I see :entity not available

**Comment by jpcornil-git** [comment]
Could you post the section of your configuration.yaml related to mpc23017 and associated customize section ?

If you rename an entity in yaml HA will also create a new one, i.e. you will have to adapt you card to use the new name (old entity won't exist/be available anymore)
